homeassistant/components/integration/sensor.py

```diff
@@ -414,19 +441,49 @@ async def async_added_to_hass(self) -> None:
             self._attr_device_class = state.attributes.get(ATTR_DEVICE_CLASS)
             self._unit_of_measurement = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
 
+        if self._max_sub_interval is not None:
+            source_state = self.hass.states.get(self._sensor_source_id)
+            self._schedule_max_sub_interval_exceeded_if_state_is_numeric(source_state)
+            self.async_on_remove(self._cancel_max_sub_interval_exceeded_callback)
+            handle_state_change = self._integrate_on_state_change_and_max_sub_interval
+        else:
+            handle_state_change = self._integrate_on_state_change_callback
+
         self.async_on_remove(
             async_track_state_change_event(
                 self.hass,
                 [self._sensor_source_id],
-                self._handle_state_change,
+                handle_state_change,
             )
         )
 
     @callback
-    def _handle_state_change(self, event: Event[EventStateChangedData]) -> None:
+    def _integrate_on_state_change_and_max_sub_interval(
+        self, event: Event[EventStateChangedData]
+    ) -> None:
+        self._cancel_max_sub_interval_exceeded_callback()
         old_state = event.data["old_state"]
         new_state = event.data["new_state"]
+        try:
+            self._integrate_on_state_change(old_state, new_state)
+            self._last_integration_trigger = _IntegrationTrigger.StateChange
+            self._last_integration_time = datetime.now(tz=UTC)
+        finally:
+            # if max_sub_interval exceeds, by construction there was no state change, the source is assumed constant new_state over max_sub_interval
```

Comment:
I don't really understand this comment, can you try to rephrase it, while also adding line breaks to keep the lines at or below 89 characters?